src/main/java/org/humancellatlas/ingest/project/Project.java

```diff
@@ -67,4 +64,18 @@ public SubmissionEnvelope getOpenSubmissionEnvelope(){
         }
         return null;
     }
-}
\ No newline at end of file
+
+    @JsonIgnore
+    public boolean removeSubmissionEnvelope(SubmissionEnvelope submissionEnvelope) {
```

Comment:
You could split removal of submission envelope and removing the supplementary files into 2 functions.
Or just rename this removeSubmissionEnvelopeData just to be clear that this isn't only removing the submission envelope but also its data linked which is linked to this project.